daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -304,17 +310,21 @@ abstract class SequenceParserBase(
         }
         case _: FailedParseAttemptStatus => { // MissingSeparator with failure will match here
           Assert.invariant(pstate.isFailure)
-          if (hasPoU && !wasDescriminatorSet &&
+          if (hasPoU && !wasDiscriminatorSet &&
             (roStatus.isInstanceOf[RequiredOptionalStatus.Optional])) {
             // we back up and finish the array at the prior element if any.
+            val cause = pstate.processorStatus.asInstanceOf[Failure].cause
```

Comment:
Unused.

Reply:
That was left over from some debugging.  It is removed now.